icicle-compiler/main/icicle.hs

```diff
@@ -269,7 +279,19 @@ runQuery f msrc query = do
   let compSecs = realToFrac (queryCompilationTime query) :: Double
   liftIO (printf "icicle: compilation time = %.2fs\n" compSecs)
 
-  liftIO (maybe (return ()) (flip writeUtf8 (querySource query)) msrc)
+  case msrc of
+    Nothing ->
+      pure ()
+    Just src ->
+      writeUtf8 src (querySource query)
+
+  case sfOutputSchema (queryFleet query) of
+    Nothing ->
+      pure ()
```

Comment:
It will be `Nothing` when we're doing sparse output.

We always output a schema if we're doing dense, and never for sparse. The `--output-schema` option is really just to override the default path that the schema is written to. I can see what you're saying, just not sure how to surface it.